src/App.tsx

```diff
@@ -3,7 +3,34 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+type Movie = {
+  title: string
+  description: string
+  imgUrl: string
+  imdbUrl: string
+  imdbId: string
+};
+
 export const App: React.FC = () => {
+  const [visibleMovies, setVisibleMovies] = React.useState(moviesFromServer);
+  const [query, setQuery] = React.useState('');
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value.trim().toLowerCase());
+  };
+
+  React.useEffect(() => {
```

Comment:
It's better to have a named import for any react hooks.
```
import { useEffect } from 'react';

...

useEffect(() => {
  ...
```

Because if you will have more complex components, your code will be worse readable (you will use too many `React.` repetitions).